api/envoy/extensions/filters/http/dynamic_forward_proxy/v3/dynamic_forward_proxy.proto

```diff
@@ -74,6 +74,14 @@ message PerRouteConfig {
     //
     //   If the header appears multiple times only the first value is used.
     string host_rewrite_header = 2;
+
+    // Specifies a hostname to be used only for DNS resolution, while preserving the original host header for
+    // forwarding. If not set or empty, the original host header value will be used for DNS lookup.
+    //
+    // Note: Unlike ``host_rewrite_literal`` and ``host_rewrite_header``, this option only affects the DNS resolution
+    // process and does not modify the host header used for forwarding. This makes it safe to use in combination with
+    // HCM host rewrite options.
```

Comment:
Can you comment example use cases of this feature?

It sounds like this will result in a host say www.foo.com being inserted into the DNS cache and DFP cluster with an address for www.bar.com?

if so this sounds potentially dangerous, because one downstream listener can "corrupt" a shared DFP cluster with addresses not aligned with the host.

Would it work to have a filter ahead of your DFP filter setting Http11ProxyInfo (which could possibly be renamed for new use cases).  I think this struct largely does what you want - if you have a Http11ProxyInfo for bar.com the DFP filter would do a lookup for bar.com and insert host foo.com with no associated address, then when that particular stream creates an upstream connection it will be logically assocated with foo.com but connect to the IP addresses in for bar.com
/wait-any

Reply:
@alyssawilk Thanks for taking a look and apologies for the delayed response. The use-case is pretty similar to [this issue](https://github.com/envoyproxy/envoy/issues/37644).

Any other suggestions on how we can do this safely?

Reply:
yeah I think I understand what you're trying to do I guess my question is why do you want to DNS resolve and send data to a different host than is in the header?  I'm not clear what the use case is in the issue either, but if I understand it better I may be able to advise on a workaround.
/wait-any